dotCMS/src/integration-test/java/com/dotmarketing/portlets/workflows/actionlet/PushNowActionletTest.java

```diff
@@ -0,0 +1,215 @@
+    }
+
+    private static void createFilterDescriptor(final String key, final boolean defaultFilter){
+        final Map<String,Object> filtersMap = new HashMap<>();
+        APILocator.getPublisherAPI().addFilterDescriptor(new FilterDescriptor(key,key,filtersMap,defaultFilter,"DOTCMS_BACK_END_USER"));
+    }
+
+    private static Environment createEnvironment (final String name) throws DotDataException, DotSecurityException {
+
+        final EnvironmentAPI environmentAPI = APILocator.getEnvironmentAPI();
+        final Environment environment = new Environment();
+        final List<Permission> permissions = new ArrayList<>();
+        permissions.add(new Permission(environment.getId(),
+                APILocator.getRoleAPI().loadRoleByKey(adminUser.getUserId()).getId(),
+                PermissionAPI.PERMISSION_USE));
+
+        environment.setName(name);
+        environment.setPushToAll(false);
+        environmentAPI.saveEnvironment(environment, permissions);
+
+        return environment;
+    }
+
+    /**
+     * This test is for checking if the Push Now Actionlet creates the Bundle with the filter selected.
+     * When the Workflow with the sub-action is invoked it creates a Bundle to PP the contentlet,
+     * that Bundle that is created must have the selected filter as filter Key.
+     */
+    @Test
+    public void test_PushNowActionlet_BundleUsesFilterKeySet()
+            throws DotSecurityException, DotDataException, IllegalAccessException, AlreadyExistException, InstantiationException {
+        //Create Content Type
+        createTestContentType();
+        //Create Filter
+        final String defaultFilterKey = "testDefaultFilterKey.yml"+System.currentTimeMillis();
+        createFilterDescriptor(defaultFilterKey,true);
+        final String filterKey = "testFilterKey.yml"+System.currentTimeMillis();
+        createFilterDescriptor(filterKey,false);
+        //Create Environment
+        final Environment environment = createEnvironment("TestEnvironment_" + System.currentTimeMillis());
+        //Create Workflow and pass the env and the filterKey
+        createWorkflowWithPushNowActionlet(environment.getName(),filterKey);
+        //Create Contentlet
+        final Contentlet cont = new Contentlet();
+        cont.setContentTypeId(type.id());
+        cont.setOwner(APILocator.systemUser().toString());
+        cont.setModDate(new Date());
+        cont.setLanguageId(1);
+        cont.setStringProperty("title", "Test Save");
+        cont.setStringProperty("txt", "Test Save Text");
+        cont.setIndexPolicy(IndexPolicy.WAIT_FOR);
+        cont.setBoolProperty(Contentlet.IS_TEST_MODE, true);
+        Contentlet contentlet1 = contentletAPI.checkin(cont, adminUser, false);
+
+        // Set the appropriate workflow action to the contentlet
+        contentlet1.setActionId(
+                schemeStepActionResult.getAction().getId());
+
+        // Triggering the actionlets
+        WorkflowProcessor processor =
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'processor' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5458951)